lib/semian/lru_hash.rb

```diff
@@ -0,0 +1,63 @@
+class LRUHash
+  extend Forwardable
+  def_delegators :@table, :size, :count, :empty?, :keys, :clear, :delete
+  attr_reader :table, :minimum_time_in_lru
+
+  class NoopMutex
+    def synchronize(*)
+      yield
+    end
+  end
+
+  def initialize(minimum_time_in_lru = 300)
```

Comment:
It's fairly counterintuitive that the `minimum_time_in_lru` is the default only option to an initializer. That's going to be painful if we plan on maintaining backward compatibility in future releases. Can you make this a named argument? When I looked at this initially, I thought you'd re-introduced size.